src/Eto/Forms/ThemedControls/ThemedDocumentControlHandler.cs

```diff
@@ -573,8 +575,11 @@ void DrawTab(Graphics g, ThemedDocumentPageHandler tab, int i)
 		g.FillRectangle(backcolor, tabRect);
 		if (tab.Image != null)
 		{
-			var imageSize = tab.Image.Size;
-			g.DrawImage(tab.Image, tabRect.X + TabPadding.Left + (maxImageSize.Width - imageSize.Width) / 2, (tabDrawable.Height - imageSize.Height) / 2);
+			g.SaveTransform();
+			g.ImageInterpolation = ImageInterpolation.High;
+			var imageSize = tab.ImageSize.IsEmpty ? tab.Image.Size : tab.ImageSize;
```

Comment:
Having `ImageSize` public property and setting it to a deliberately "wrong" value can end up in less-than-beautiful results. I would argue the desired image size should be lead by the `ThemedDocumentControl` itself (it knows the dimensions of the tabs it renders and it also should know the ideal dimensions for the image inside that tab).

Shall we just scale `tab.Image` up/down to fit that ideal rectangle?

Reply:
I got your point, I will revert changes for exposing ImageSize and let you know.

Reply:
Hi @cwensley ,

- I moved the color properties to `ThemedDocumentControlHandler`

- Implemented `PageClosing` for the Gtk handler.

- Couldn't implement `HasNavigationButtons` (now called `HasNavigationButtons`) for the Gtk handler as the gtk Nootbook control doesn't allow to hide the `HasForwardStepper` and `HasBackwordStepper` because they are readonly, Hence I move the property to the `ThemedDocumentControlHandler`

Reply:
Hi @ievgen-baida ,
I have introduce a property `UseFixedTabHeight` along with a checkbox in the DocumentControlSection demo dialog. When set `UseFixedTabHeight` to true it resizes the images inside the Tab height, then we have a fixed tab height that is calculated by the font size and vertical padding.

Reply:
Looks good to me.

Only thing, I wonder if instead of relying on the font size when calculating the height of the image we would rather use some sensible standard size, like 16x16?

Reply:
@ievgen-baida, sounds like a good idea. Applied.

Reply:
Thank you. Looks good to me.